include/cuda/std/detail/libcxx/include/__config

```diff
@@ -1744,10 +1737,30 @@ extern "C" _LIBCUDACXX_FUNC_VIS void __sanitizer_annotate_contiguous_container(
 #endif
 
 // Thread API
-#if !defined(_LIBCUDACXX_HAS_NO_THREADS) && \
-    !defined(_LIBCUDACXX_HAS_THREAD_API_PTHREAD) && \
-    !defined(_LIBCUDACXX_HAS_THREAD_API_WIN32) && \
-    !defined(_LIBCUDACXX_HAS_THREAD_API_EXTERNAL)
+#ifndef _LIBCUDACXX_HAS_THREAD_API_EXTERNAL
+#if defined(_LIBCUDACXX_COMPILER_NVRTC) \
+ || defined(__EMSCRIPTEN__)
+#  define _LIBCUDACXX_HAS_THREAD_API_EXTERNAL
+#endif
+#endif // _LIBCUDACXX_HAS_THREAD_API_EXTERNAL
+
+#ifndef _LIBCUDACXX_HAS_THREAD_API_CUDA
+#if defined(__cuda_std__)
+#  define _LIBCUDACXX_HAS_THREAD_API_CUDA
+#endif // __cuda_std__
+#endif // _LIBCUDACXX_HAS_THREAD_API_CUDA
+
+#ifndef _LIBCUDACXX_HAS_THREAD_API_WIN32
+#if defined(_LIBCUDACXX_COMPILER_MSVC)        \
+ && !defined(_LIBCUDACXX_HAS_THREAD_API_CUDA)
+#  define _LIBCUDACXX_HAS_THREAD_API_WIN32
+#endif
+#endif // _LIBCUDACXX_HAS_THREAD_API_WIN32
```

Comment:
Doesn't this change what things are defined, compared to the current code? It seems to me that for MSVC, we used to... define only API_WIN32 in the host TU and only API_CUDA in the device TUs, and with this new definition we'll end up defining both for both kinds of TUs. Maybe there's no difference in how these are currently used, but I don't think this is quite correct either way; the way upstream defines them below definitely assumes that only one is defined at once. (I think we were also wrong in how we defined API_EXTERNAL earlier.)

Reply:
Yeah those changes were accidental. Good catch!

I reverted to the previous state